components/DocMenu/react/_DocLink.tsx

```diff
@@ -23,7 +23,7 @@ export const DocLink: React.FC<DocLinkProps> = ({
     if (item.active) {
       onActive?.(activeLIRef?.current?.offsetTop || 0)
     }
-  }, [])
+  })
```

Comment:
This will run on every rerender; your previous implementation was running only "on mount" (which is what happens with empty array in useEffect). 

So here you should keep the empty array; and you can eslint-ignore with a comment explaining it is meant to only happen on mount.

Reply:
Or if you want to be HYPER kosher and avoid disabling the linter, then you can do this:
```
const dependenciesRef = React.useRef({ item, onActive, activeLIRef });
// Capture the current values on every render
dependenciesRef.current = { item, onActive, activeLIRef };

React.useEffect(() => {
  const { item, onActive, activeLIRef } = dependenciesRef.current;
  if (item.active) {
    onActive?.(activeLIRef?.current?.offsetTop || 0);
  }
  // Since dependencies are in a ref, you can still use an empty dependency array
}, []);
```

But a linter disabled with a comment might be good enough here.